airbyte-integrations/connectors/source-gitlab/source_gitlab/spec.json

```diff
@@ -2,37 +2,106 @@
   "documentationUrl": "https://docs.airbyte.io/integrations/sources/gitlab",
   "connectionSpecification": {
     "$schema": "http://json-schema.org/draft-07/schema#",
-    "title": "Source Gitlab Singer Spec",
+    "title": "Source Gitlab Spec",
     "type": "object",
-    "required": ["api_url", "private_token", "start_date"],
-    "additionalProperties": false,
+    "required": ["api_url", "start_date"],
+    "additionalProperties": true,
     "properties": {
       "api_url": {
         "type": "string",
+        "title": "API URL",
         "examples": ["gitlab.com"],
+        "default": "gitlab.com",
         "description": "Please enter your basic URL from Gitlab instance"
       },
-      "private_token": {
-        "type": "string",
-        "description": "Log into your Gitlab account and then generate a personal Access Token.",
-        "airbyte_secret": true
-      },
       "groups": {
         "type": "string",
+        "title": "Groups",
         "examples": ["airbyte.io"],
         "description": "Space-delimited list of groups. e.g. airbyte.io"
       },
       "projects": {
         "type": "string",
+        "title": "Projects",
         "examples": ["airbyte.io/documentation"],
         "description": "Space-delimited list of projects. e.g. airbyte.io/documentation meltano/tap-gitlab"
       },
       "start_date": {
         "type": "string",
+        "title": "Start Date",
         "description": "The date from which you'd like to replicate data for Gitlab API, in the format YYYY-MM-DDT00:00:00Z. All data generated after this date will be replicated.",
         "examples": ["2021-03-01T00:00:00Z"],
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$"
+      },
+      "credentials": {
+        "title": "Authorization Method",
+        "type": "object",
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "OAuth2.0",
+            "required": ["client_id", "client_secret", "access_token"],
+            "properties": {
+              "credentials": {
+                "type": "string",
+                "const": "oauth2.0",
+                "enum": ["oauth2.0"],
+                "default": "oauth2.0",
+                "order": 0
+              },
```

Comment:
you can use the option `const` only because `enum`, `default` and `order` will are not relevant

Reply:
Changed.